preprocessor.py

```diff
@@ -0,0 +1,133 @@
+import re
+import sys
+
+class PreProcessor:
+  def __init__(self,url_repl='', email_repl='', phone_repl='', num_repl='', date_repl='', emoji_repl='', html_tag_repl=''):
+    self.url_repl = url_repl
+    self.email_repl = email_repl
+    self.phone_repl = phone_repl
+    self.date_repl = date_repl
+    self.num_repl = num_repl
+    self.emoji_repl = emoji_repl
+    self.html_tag_repl = html_tag_repl
+
+    self._preProccess =[]
+    self._init_preProccess()
+    
+  def _preprocess_email(self,input_text,email_repl):
+    regex_pattern = re.compile(
```

Comment:
make sure all the regexes are compiled at the init method. load regexes from an external dictionary - keep regexes as a mapping and compile during init.